src/auth/index.ts

```diff
@@ -0,0 +1,50 @@
+import { Request, Response, NextFunction } from "express";
+import { USERS_FILE_PATH } from "../constants/filePaths";
+import { readFile } from "fs/promises";
+import { RESPONSE_CODE_SERVER_ERROR } from "../constants/responseCodes";
+
+async function authenticateUser(
+  req: Request,
+  res: Response,
+  next: NextFunction
+) {
+  try {
+    const userId = req.header("x-user-id") as string;
+    if (!userId) {
+      return res.status(401).json({
+        data: null,
+        error: {
+          message: "Header x-user-id is missing",
+        },
+      });
+    }
+
+    const usersData = await readFile(USERS_FILE_PATH, "utf8");
+    const users = JSON.parse(usersData);
+
+    const foundUser = users.find((user: { id: string }) => user.id === userId);
+
+    if (!foundUser) {
+      return res.status(401).json({
+        data: null,
+        error: {
+          message: "No user with such id",
+        },
+      });
+    }
+
+    (req as any).user = foundUser;
```

Comment:
if you wish to use strict typescript - you can declare you own Request interface (via extending the express interface). In this case you no need each time to use "as any"